modules/geo-layers/src/mvt-layer/mvt-layer.ts

```diff
@@ -22,28 +38,80 @@ const defaultProps = {
   binary: true
 };
 
-export default class MVTLayer extends TileLayer {
-  initializeState() {
+export type TileJson = {
+  tilejson: string;
+  tiles: string[];
+  // eslint-disable-next-line camelcase
+  vector_layers: any[];
+  attribution?: string;
+  scheme?: string;
+  maxzoom?: number;
+  minzoom?: number;
+  version?: string;
+};
+
+/** All props supported by the MVTLayer */
+export type MVTLayerProps<DataT extends Feature = Feature> = _MVTLayerProps &
+  GeoJsonLayerProps<DataT> &
+  TileLayerProps<DataT>;
+
+/** Props added by the MVTLayer  */
+export type _MVTLayerProps = {
+  /** Called if `data` is a TileJSON URL when it is successfully fetched. */
+  onDataLoad?: ((tilejson: TileJson | null) => void) | null;
+
+  /** Needed for highlighting a feature split across two or more tiles. */
+  uniqueIdProperty?: string;
+
+  /** A feature with ID corresponding to the supplied value will be highlighted. */
+  highlightedFeatureId?: string | null;
+
+  /**
+   * Use tile data in binary format.
+   *
+   * @default true
+   */
+  binary?: boolean;
+
+  /**
+   * Loaders used to transform tiles into `data` property passed to `renderSubLayers`.
+   *
+   * @default [MVTWorkerLoader] from `@loaders.gl/mvt`
+   */
+  loaders?: Loader[];
+};
+
+type ContentWGS84Cache = {_contentWGS84?: Feature[]};
+
+export default class MVTLayer<DataT extends Feature = Feature, ExtraProps = {}> extends TileLayer<
+  DataT,
+  Required<_MVTLayerProps> & ExtraProps
+> {
+  static layerName = 'MVTLayer';
+  static defaultProps = defaultProps;
+
+  initializeState(): void {
     super.initializeState();
     // GlobeView doesn't work well with binary data
-    const binary = this.context.viewport.resolution !== undefined ? false : this.props.binary;
+    const binary = this.context.viewport instanceof _GlobeViewport ? false : this.props.binary;
```

Comment:
Should not check for `GlobeViewport` here - we may add other projections soon. You can add a `resolution` declaration to the `Viewport` class, otherwise disable binary when `viewport.isGeospatial && !(viewport instanceof WebMercatorViewport)`

Reply:
Was worried about correctness so, went back to `resolution` check. It's _only_ conversion to TS anyways...